sqrl-server/sqrl-server-vertx/src/main/java/com/datasqrl/graphql/postgres_log/PostgresListenNotifyConsumer.java

```diff
@@ -0,0 +1,111 @@
+import io.vertx.sqlclient.PreparedQuery;
+import io.vertx.sqlclient.Row;
+import io.vertx.sqlclient.RowSet;
+import io.vertx.sqlclient.Tuple;
+import java.time.OffsetDateTime;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Consumer;
+import lombok.AllArgsConstructor;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@AllArgsConstructor
+public class PostgresListenNotifyConsumer {
+
+  private VertxJdbcClient sqlClient;
+  private String listenQuery;
+  private String onNotifyQuery;
+  private List<String> parameters;
+  private Vertx vertx;
+  private PgConnectOptions pgConnectOptions;
+
+  @SneakyThrows
+  public void subscribe(Consumer<Object> listener) {
+    // Establish a direct PgConnection
+    PgConnection.connect(vertx, pgConnectOptions, res -> {
+      if (res.succeeded()) {
+        PgConnection pgConnection = res.result();
+
+        // Set the notification handler
+        pgConnection.notificationHandler(notification -> {
+          log.trace("Received notification on channel: {} Payload: {}",
+              notification.getChannel(), notification.getPayload());
+
+          handleNotification(notification, listener);
+        });
+
+        // Execute the LISTEN command to listen for notifications on a specific channel
+        pgConnection.query(listenQuery).execute(queryResult -> {
+          if (queryResult.succeeded()) {
+            log.info("LISTEN command executed successfully: {}", listenQuery);
+          } else {
+            log.error(String.format("Unable to execute LISTEN command: %s", queryResult.cause().getMessage()), queryResult.cause());
+          }
+        });
+      } else {
+        log.error(String.format("Unable to establish connection. %s", res.cause().getMessage()), res.cause());
+      }
+    });
+  }
+
+  public void handleNotification(PgNotification notification, Consumer<Object> listener) {
+    JsonObject jsonPayload = new JsonObject(notification.getPayload());
+
+    List<Object> paramObj = new ArrayList<>();
+
+    // TODO (Soma) - We need to properly handle the payload types here.
+    //  This is hard coded currently and tailored to the sensors example.
+    for (String parameter : parameters) {
+      if (parameter.equals("timeSec")) {
+        paramObj.add(OffsetDateTime.parse(jsonPayload.getString(parameter)));
+
+      } else if (parameter.equals("sensorid")) {
+        paramObj.add(jsonPayload.getInteger(parameter));
+      } else {
+        throw new IllegalArgumentException("Unknown parameter: " + parameter);
+      }
+    }
+
+    // TODO (Soma) - It feels odd that we are using vertx a bit differently compared to how we
+    //  handle the notifications. This is the accepted way of running queries in the codebase
+    //  however in case on notifications we are forced to use PGConnection since that's the
+    //  only way currently to listen to notifications.
```

Comment:
This feels fine. We have an explicit connection to listen to notifications (only would be a few connections since its shared), and then use the connection pooler to do the queries. 

So are we executing these queries regardless if we have a listener? Does that make sense to do?

Reply:
> This feels fine. We have an explicit connection to listen to notifications (only would be a few connections since its shared), and then use the connection pooler to do the queries.
Ok.

> So are we executing these queries regardless if we have a listener? Does that make sense to do?
No, we are only executing the onNotify queries when a notification arrives. But I'm not sure what do you want to hint with this question. Am I missing something?

Reply:
High level there are two approaches:
- We have one database connection per subscription type and we use flux + calcite or trivial filtering logic to fan out all of the requests
- We create a new database connection per subscription _connection_. We push down filters into the database and/or use calcite/trivial filters.

I'm just following up with what approach we are using and if there will be queries against the database if there are no subscription connections.

Reply:
So I believe for both options we need to have at least 2 connections. We need to keep one connection open and dedicated to receiving notifications. You cannot reuse it for other operations while you are listening on it.

Apart from that, do I understand correctly that this topic is not a blocker on merging this PR?